app/models/katello/concerns/redhat_extensions.rb

```diff
@@ -44,16 +44,63 @@ def construct_name(family)
       end
 
       def medium_uri_with_content_uri(host, url = nil)
-        if host.try(:content_source) && (repo_details = kickstart_repo(host))
-          URI.parse(repo_details[:path])
+        if host.try(:content_source) && (repos = distribution_repositories(host)).present?
+          if repos.size == 1
+            # There is exactly 1 ks repo distribution matching the os, arch, release
+            # cv, env for this host. Just get the media path from available repo.
+            URI.parse(repos.first.full_path(host.content_source))
```

Comment:
Thats an interesting idea. But lemme explain something. Look at the image below
![OS Host](http://i.imgur.com/OirIlEz.png)
See how the media got auto populated.
So the UI interaction does not give the user a choice in selecting the media. If the user chose RedHat 7.2 as os, x86_64, content source, env, cv and there is only one matching media. So in theory we donot need to validate here.